### ISSUE TITLE
Address isn't understood

### ISSUE THREAD

**Issue by Scrxtchy** [JustOff/ca-archive#2]
Using Latest Waterfox, pressing the extension button brings me to this page

![image](https://user-images.githubusercontent.com/12771982/32690383-fb4ef322-c749-11e7-8d9d-971c511a702f.png)

**Comment by Scrxtchy** [comment]
Mitigated locally by navigating to chrome://ca-archive/content/ca-archive.html

**Comment by JustOff** [comment]
You probably forced e10s on, unfortunately CAA can't run in this mode for now.

**Comment by JustOff** [comment]
I'll try to get it work with e10s, but at the moment it's obviously not my top priority.

**Comment by mparnelldmp** [comment]
In Waterfox, at least, this is an option for now:

![image](https://user-images.githubusercontent.com/25654372/32843041-571246f4-c9e4-11e7-8d06-81a6db7750f6.png)

**Comment by JustOff** [comment]
It seems Waterfox specific feature, patches to run CAA in non-e10 window are welcome.

**Comment by grahamperrin** [comment]
As a side note, e10s is enabled by default in Waterfox 56.0 and not automatically disabled when incompatible extensions are enabled.

**Comment by grahamperrin** [comment]
> … patches to run CAA in non-e10 window are welcome.

I'd love to contribute but (sorry) no relevant coding skills. 

@MrAlex94 please: for the next release of Waterfox (56.2.4, I guess) do we need to seek additional help with this issue?

**Comment by grahamperrin** [comment]
### Workaround

For cases where a **non**-e10s window of Waterfox in multi-process mode does **not** allow use of version 1.2.0 of the extension. 

Whilst there _is_ a simpler workaround (bear with me – more to follow), stepping through what's below might help towards an eventual fix. 

1. Start Waterfox in multi-process mode
2. remove version 1.2.0
3. download and extract https://github.com/JustOff/ca-archive/releases/tag/1.0.7 (avoid the e10s warning that was added to 1.0.8)
4. https://github.com/JustOff/ca-archive/releases/tag/1.2.0
5. save the `.xpi` (refrain from addition)
6. **about:debugging#addons**
7. ☑ Enable add-on debugging
8. Load temporary add-on
9. open the extracted `bootstrap.js` for 1.0.7
10. open a new non-e10s window
11. click the Classic Add-ons Archive button
12. ignore the invitation to reload the page
13. bring to front the **about:debugging#addons** tab
14. remove 1.0.7
15. add, or temporarily add, 1.2.0
16. bring to front the **caa:** tab
17. reload
18. again. reload.

**Comment by JustOff** [comment]
That is not a good workaround. Please try [CAA 1.2.1b1](https://github.com/JustOff/ca-archive/releases/download/1.2.0/ca-archive-1.2.1b1.xpi), which tries to run in a non-e10s window.

**Comment by grahamperrin** [comment]
Thanks! 

1.2.1b1 is much better. Hopefully no longer any situation in which the modal dialogue will appear. 

### A couple of edge cases

#### 1

A special URL e.g. `about:`… as the first and only tab in the only window when Waterfox starts. 

#### 2

A preference to _Show your windows and tabs from last time_ when Waterfox starts, with an extension such as _Speed Dial_ caa:addon/speed-dial affecting new tab behaviour, and with the affected tab frontmost in a sole window before quit: 

![2018-10-07 22 44 23 undefined](https://user-images.githubusercontent.com/192271/46587422-131d9600-ca84-11e8-85f0-0965c9b382a0.png)

----

In cases such as these there'll be the generic information page: 

> The address wasn't understood

– with a tab context menu workaround: 

![2018-10-07 22 45 38 undefined](https://user-images.githubusercontent.com/192271/46587424-187ae080-ca84-11e8-824c-2915c155918d.png)

Alternatively: 

1. close the caa: tab
2. bring to front any tab with a normal (e.g. http:// or https://) address
3. reuse the Classic Add-ons Archive button.

**Comment by JustOff** [comment]
The problem here is that Waterfox does not restore the non-e10s status of the windows after restart, so they become unsuitable for `caa:` scheme. This is pretty sad and now I don’t even know if it's worth using non-e10s windows at all or is't better to return the requirement to disable multi-process mode so as not to confuse anyone with the issue after restarting.

**Comment by JustOff** [comment]
Please try [CAA 1.2.1b3](https://github.com/JustOff/ca-archive/releases/download/1.2.0/ca-archive-1.2.1b3.xpi).

**Comment by grahamperrin** [comment]
I seem to get mixed results with 1.2.1b3. 

The first impression was good; it was easy to get a window with caa: working. 

Later use of the same profile was not easy. I could get only the modal dialogue, with its insistence upon disabling multi-process. 

(Maybe the first impression was based on me aiming immediately for the workaround, in response to the generic _… not understood …_ text. Honestly I can't recall exactly what I did immediately at the start of the test.) 

Does 1.2.1b3 lack the (1.2.1b1) ability to automatically open a new non-e10s window?

### Wording for the non-modal red alert at caa:

Instead of: 

> Waterfox in multi-process mode can't restore caa: urls after restart!

– maybe this: 

> Multi-process Waterfox does not save non-e10s state when saving sessions

**Comment by JustOff** [comment]
>  I could get only the modal dialogue, with its insistence upon disabling multi-process.

This modal dialog is called from the block `if (mode == e10s && appname != "Waterfox")`; do you use any extensions or other ways to hide or fake the application name?

> Does 1.2.1b3 lack the (1.2.1b1) ability to automatically open a new non-e10s window?

All 1.2.1bX versions have this ability, they differ in the manner in which the e10s mode is detected and in the presence of a non-modal warning.

> Wording for the non-modal red alert at caa:

Sorry, but your wording is too general to be useful for the average user.

**Comment by grahamperrin** [comment]
> … do you use any extensions or other ways to hide or fake the application name?

I'm testing in a profile that has just the one extension. 

![2018-10-11 20 06 23 javascript application](https://user-images.githubusercontent.com/192271/46827678-2cc72200-cd91-11e8-8366-e39415413a27.png)

**Comment by JustOff** [comment]
Then I absolutely don't understand how you can see *"Multi-process mode is not supported"* in 1.2.1b1-b3 :panda_face:

**Comment by JustOff** [comment]
@grahamperrin, if you can still reproduce the issue with *"Multi-process mode is not supported"* is displayed, please email me a copy of the test profile where it's observed (you can reduce the size of the profile by excluding the ca-archive folder from it).

**Comment by grahamperrin** [comment]
Here it was, the modal, with 1.2.1b3 a few minutes ago (two frames from a screen recording, three seconds apart): 

![2018-10-12 21 05 frame](https://user-images.githubusercontent.com/192271/46893533-2a37fb80-ce69-11e8-9543-944c650dba33.png)

![2018-10-12 21 08 frame](https://user-images.githubusercontent.com/192271/46893537-315f0980-ce69-11e8-8475-db3cfe77885b.png)

After removing the extension, quitting, starting, adding the same version: behaviour was quite different – much better, of course :-)

Mysterious. 

Unfortunately (sorry) I didn't set aside a copy of the affected profile before working around the mystery but as seen in the second shot above, it was very clean.

**Comment by JustOff** [comment]
This drives me crazy, `2+2=5` :anguished:

Here is [CAA 1.2.1b4](https://github.com/JustOff/ca-archive/releases/download/1.2.0/ca-archive-1.2.1b4.xpi), which displays appname instead of "the browser" in the modal dialogue:

```
if (e10s && Services.appinfo.name != "Waterfox") {
  Services.prompt.alert(null, "Classic Add-ons Archive", 
  "Multi-process mode is not supported now,\nplease disable it and restart " + 
  Services.appinfo.name + ".");
  return;
}
```

**Comment by grahamperrin** [comment]
Well … (if we treat the first screenshot in my previous comment as inexplicable) I can not find any situation in which the modal dialogue appears. 

Am I missing something?

**Comment by JustOff** [comment]
This dialogue should never appear in Waterfox, I just don't like to leave inexplicable things unexplained.

**Comment by grahamperrin** [comment]
> This dialogue should never appear in Waterfox, …

OK :+1:

**Comment by JustOff** [comment]
I'm going to release 1.2.1 tonight to verify the e10s workaround in the wild.

**Comment by grahamperrin** [comment]
Yep, it feels release-ready to me. Wild time. Thanks. 

----

Back to the opening post, the generic message: 

> ## The address wasn’t understood

For https://github.com/JustOff/ca-archive/issues/2#issuecomment-427793921 above and the plain English red alert –

> Waterfox in multi-process mode can't restore caa: urls after restart!

– I raised the issue in the Waterfox repo: 

- [Multi-process Waterfox does not save non-e10s state when saving sessions · Issue #761 …](https://github.com/MrAlex94/Waterfox/issues/761)

**Comment by BGMcoder** [comment]
Yes, I have the same problem.  CAA gives a "The address wasn’t understood" message in Waterfox 56.2.8 (64-bit).  TMP installed and e10's is enabled.  Just confirming.
